awsclilinter/awsclilinter/cli.py

```diff
@@ -75,22 +83,89 @@ def display_finding(finding: LintFinding, index: int, total: int, script_content
             print(line, end="")
 
 
-def interactive_mode(findings: List[LintFinding], script_content: str) -> List[LintFinding]:
-    """Run interactive mode and return accepted findings."""
-    accepted = []
-    for i, finding in enumerate(findings, 1):
-        display_finding(finding, i, len(findings), script_content)
-        choice = get_user_choice("\nApply this fix? [y]es, [n]o, [u]pdate all, [s]ave and exit: ")
+def apply_all_fixes(
+    findings_with_rules: List[Tuple[LintFinding, LintRule]],
+    ast: SgRoot,
+) -> str:
+    """Apply all fixes using rule-by-rule processing.
+
+    Since multiple rules can target the same command, we must process one rule
+    at a time and re-parse the updated script between rules to get fresh Edit objects.
+
+    Args:
+        findings_with_rules: List of findings and their rules.
+        ast: Current script represented as an AST.
+
+    Returns:
+        Modified script content
+    """
+    current_ast = ast
+
+    # Group findings by rule
+    findings_by_rule: Dict[str, List[LintFinding]] = {}
+    for finding, rule in findings_with_rules:
+        if rule.name not in findings_by_rule:
```

Comment:
nit: could use `collections.defaultdict` to reduce checking here, not a strong preference.